### ISSUE TITLE
Questions regarding usecase - software defined radio/radio comms

### ISSUE THREAD

**Issue by OE3JGW** [cyberofficial/Synthalingua#150]
**First of all.. thank you. This tool is AMAZING.** 
My usecase is software defined radio, its interesting to use it for translation radio comms. 
Main issue there is the very bad audio quality sometimes, its still works good i must say. 

But sometimes i get this, im new to this so im not sure how to interpret this:
Given groups=1, weight of size [512, 80, 3], expected input[1, 128, 3000] to have 80 channels, but got 128 channels instead

Besides that, any advice on settings for bad source audio quality to get the most out of it? (radio comms)

**Comment by cyberofficial** [comment]
Hello! Can you tell me the command arguments being used?

**Comment by OE3JGW** [comment]
sure!
at first i used it with 12gb-v3, it might came from that. i use a 1080ti so if 12gb-v3 refers to vram maybe this caused it. 
i switched to 6gb and didnt had the problem again - but it was not all the time so yeah.. ill  monitor it. 
most of the time i dont know which language it is so i tested the autolock feature but it kept saying it detected en again and again and again (even after idk 30 tries or something), so i removed it. 

even tough i had some issues setting it up etc, this tool is awesome, you did an amazing job!


**atm im using this:**

@echo off
cls
call "data_whisper\Scripts\activate.bat"
call ffmpeg_path.bat
python transcribe_audio.py --ram 6gb --microphone_enabled True --set_microphone 2 --translate --target_language en --auto_model_swap
pause

**Comment by cyberofficial** [comment]
I might need to change a few things around in the code, I'll experiment with the 12gb models. I might've missed a few compatibility snippets of code for the gtx architecture. I'll need to pull out my 1080 from my closet and set up an older system to try things out. I'm working with RTX so it does make me miss older things when coding. But reports likes these help out in the long run. The 1080 should be supported and the ti version of it. I was able to run it on the 1660 cards as well as the 950 for memes (super sluggish of course)

**Comment by OE3JGW** [comment]
that was the intended purpose, i want to help make it even better :) 
yeah the 1080ti has some years on it.. 
but it works! 

would the force ram feature help using the full potential or is the force ram refering to system ram and not vram?

any advise on settings for bad quality source audio (radio comms)? 
im aware of the limitations bad audio quality (other crackling noises,etc etc) do have - im shocked it even works lol
i tested it with some pretty nasty audio and it still came up with "some" things (surely not perfect but who can ask for that haha)

**Comment by cyberofficial** [comment]
> would the force ram feature help using the full potential or is the force ram refering to system ram and not vram?

This feature is trying to force the model to load depsite low vram or ram. For example if you have 12GB GPU, the 12GB-v2 model needs about 11.5GB of Vram, so sometimes it wont load, so forcing it to load into ram might work, leaving the OS with .5GB of spare vram. its not really recommend to use unless you for sure you can squeze the system memory just right.

> any advise on settings for bad quality source audio (radio comms)?

Using audio sound compression like RTX voice from nvidia could probably also work. I think it works on GTX cards too. Some sort of sound suppressor should improve the application. Might be able to probably find free VST software to supress static sound like remove static like rtx voice using the cpu or gpu or both.

**Comment by OE3JGW** [comment]
so nothing specific for the arguments. okey :)
ill try that voice tool from nvidia, sounds interesting.

thanks for the infos, then ill stick to the 6gb model i guess ^^

**Comment by cyberofficial** [comment]
The auto model feature was removed for the time being, the currentt implementation will need to be adjusted which will come at a later date, the current version was just a hacky bandaid thing I made. I will be re-adding it back when I have a better way of handling on the fly model updates. The main purpose of the auto model swap was just to have something to rapidly change from non en models to en and back; but the current way I coded it doesn't have the correct method. It's something I need to come back and look at. I'm closing this thread as that was the main missue. Apologies for this.
